packages/plugin/src/generators/generator/generator.ts

```diff
@@ -140,6 +146,22 @@ async function updateGeneratorJson(host: Tree, options: NormalizedSchema) {
       options.skipFormat
     );
   }
+  // add dependencies
+  updateJson<PackageJson>(
```

Comment:
Wouldn't nx and devkit already exist if users have `@nx/plugin` installed? And SWC packages may not be needed if they use tsc or another bunder.

Reply:
- They would be installed but as a transitive dependency. Our generated `generator` uses them directly so we need to reference them. Not to mention, using transitive deps does not work with `PnP`, `pnpm` (under some conditions), or the `no-hoist` flag.
- This is due to what our generator generates. The purpose is to not have broken `lint` after running the generator. 

If you look at our E2E test, it calls the following commands:
- nx generate @nx/plugin:plugin ${plugin} --linter=eslint
- nx generate @nx/plugin:generator ${generator} --project=${plugin}
- nx lint ${plugin}

Despite `compiler` being set by default to `tsc`, our helper functions check returns `swc`. Perhaps there's a bug elsewhere. I will look into this on Monday to be sure.

Reply:
The previously generated package.json was broken so it falsy requested `@swc`. After last commit from @jaysoo  the tests started failing as expected.

Reply:
Should be fixed now